### PR TITLE
Bug 1852047: daemon: Remove "compare digest" code for OS updates

### PR DESCRIPTION
**Pull request by cgwalters**
We don't need to do this special casing anymore.  This
corner case only arises for the synthetic `e2e-gcp-upgrade`
test, and we actually want to test the OS update path there.

Closes: https://github.com/openshift/machine-config-operator/issues/1964

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -1351,7 +1351,10 @@ func getRefDigest(ref string) (string, error) {
 	return canon.Digest().String(), nil
 }
 
-// compareOSImageURL is the backend for checkOS.
+// compareOSImageURL is the backend for checkOS.  This function
+// used to have more sophistication around comparing the digests,
+// but we removed that.  The dummy:// could also likely go away.
+// https://github.com/openshift/machine-config-operator/issues/1964
 func compareOSImageURL(current, desired string) (bool, error) {
```
Comment (ericavonb) [inline pkg/daemon/daemon.go:1358]: We can remove the error return from the function altogether at this point.
```golang
func compareOSImageURL(current, desired string) bool {
	return desired == "" || desired == "://dummy" || current == desired
```
("No target" log omitted for demonstration only)

Comment (cgwalters) [reply pkg/daemon/daemon.go]: Yep you're right, it turns into a bigger patch but as long as we're here we might as well do the bigger cleanup.

**Comment by ericavonb** [comment]
yay for simpilified code!
/lgtm

**Comment by sinnykumari** [comment]
I am not 100% sure but there is lot  of possibility that this PR merge will create  merge conflict to the extensions feature https://github.com/openshift/machine-config-operator/pull/1941/files#diff-06961b075f1753956d802ba954d2cfb5L1319 . It will be good to fix this on top of #1941.

Adding hold to get Pr#1941 merged.
/hold

**Comment by sinnykumari** [comment]
#1941 merged
/hold cancel

**Comment by cgwalters** [comment]
Turned out the relevant bit here was already removed in https://github.com/openshift/machine-config-operator/pull/1941
